source/common/upstream/upstream_impl.cc

```diff
@@ -340,6 +367,30 @@ ClusterInfoImpl::ClusterInfoImpl(const envoy::api::v2::Cluster& config,
     idle_timeout_ = std::chrono::milliseconds(
         DurationUtil::durationToMilliseconds(config.common_http_protocol_options().idle_timeout()));
   }
+
+  auto filters = config.filters();
```

Comment:
This looks very similar to the code in listemer_manager_impl. Dedupe?

Reply:
Yep, cut-and-paste. It should be pulled up to a common place. Thought I had a TODO on that one, well spotted.